src/components/views/messages/ReactionsRowButton.tsx

```diff
@@ -105,22 +108,56 @@ export default class ReactionsRowButton extends React.PureComponent<IProps, ISta
         }
 
         const room = this.context.getRoom(mxEvent.getRoomId());
+        const customReactionImagesEnabled = SettingsStore.getValue("feature_render_reaction_images");
         let label: string | undefined;
+        let customReactionName: string | undefined;
         if (room) {
             const senders: string[] = [];
             for (const reactionEvent of reactionEvents) {
                 const member = room.getMember(reactionEvent.getSender()!);
                 senders.push(member?.name || reactionEvent.getSender()!);
+                if (
+                    customReactionImagesEnabled &&
+                    (reactionEvent.event.content?.[REACTION_SHORTCODE_KEY.name] ||
+                        reactionEvent.event.content?.[REACTION_SHORTCODE_KEY.altName])
+                ) {
+                    customReactionName =
+                        reactionEvent.event.content?.[REACTION_SHORTCODE_KEY.name] ||
+                        reactionEvent.event.content?.[REACTION_SHORTCODE_KEY.altName];
+                }
             }
 
             const reactors = formatCommaSeparatedList(senders, 6);
             if (content) {
-                label = _t("%(reactors)s reacted with %(content)s", { reactors, content });
+                label = _t("%(reactors)s reacted with %(content)s", {
+                    reactors,
+                    content: customReactionName || content,
+                });
             } else {
                 label = reactors;
             }
         }
 
+        let reactionContent = (
+            <span className="mx_ReactionsRowButton_content" aria-hidden="true">
+                {content}
+            </span>
+        );
+        if (customReactionImagesEnabled && content.startsWith("mxc://")) {
+            const imageSrc = mediaFromMxc(content).srcHttp;
+            if (imageSrc) {
+                reactionContent = (
+                    <img
+                        className="mx_ReactionsRowButton_content"
+                        alt={customReactionName || content}
```

Comment:
`content` being an `mxc://` URL is not a helpful alt text

Reply:
what would you recommend the alt text be? Just empty in the case of no custom reaction name?

Reply:
`_("Custom reaction")` perhaps? its the edge case given the vast majority will have shortcodes, no?

Reply:
sounds good. I guess it will probably be an edge case.